android/src/main/java/com/rnfs/RNFSManager.java

```diff
@@ -156,6 +156,37 @@ public void readFileAssets(String filepath, Promise promise) {
   }
 
   @ReactMethod
+  public void readFileRes(String filename, Promise promise) {
+    String prefix = filename.substring(filename.lastIndexOf(".")+1);
+    String name = filename.substring(0, filename.lastIndexOf("."));
+    Boolean isImage = prefix.equals("png") || prefix.equals("jpg") || prefix.equals("jpeg") || prefix.equals("bmp") || prefix.equals("gif") || prefix.equals("webp") || prefix.equals("psd") || prefix.equals("svg") || prefix.equals("tiff");
+    InputStream stream = null;
+    try {
+      int res = getReactApplicationContext().getResources().getIdentifier(name, isImage ? "drawable" : "raw", getReactApplicationContext().getPackageName());
```

Comment:
until this line and everything before "try" and after "catch" is the same in readFileRes, copyFileRes and existsRes. I understand the purpose, but wouldn't it make sense to isolate all this boilerplate in a separated private method and give it a callback what to do with it finally (reading, copying, existance-check) in order to reduce duplicated code and make it more clean